numba/targets/arraymath.py

```diff
@@ -415,6 +415,22 @@ def array_min_impl(arry):
                         min_value = v
             return min_value
 
+    elif isinstance(ty, types.Float):
+        def array_min_impl(arry):
+            if arry.size == 0:
+                raise ValueError(MSG)
+
+            it = np.nditer(arry)
+            min_value = next(it).take(0)
```

Comment:
If `arry[0]` is `NaN` such that `min_value` has the value `NaN` at this point the following loop need not proceed?

Reply:
Yes, good point. Returning at that point will help in the leading-`nan` case. I'll update accordingly